tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,204 @@
+        writer = tf.summary.FileWriter(dir, sess.graph)
+        for i in range(40):
+            with sess.as_default():
+                summary, _ = sess.run([merged, total])
+            writer.add_summary(summary, global_step=i)
+        shutil.rmtree(dir)
+        writer.close()
+
+    return client.get_run(run.info.run_id)
+
+
+def test_tf_core_autolog_logs_scalars(tf_core_random_tensors):
+    assert 'a' in tf_core_random_tensors.data.metrics
+    assert tf_core_random_tensors.data.metrics['a'] == 3.0
+    assert 'b' in tf_core_random_tensors.data.metrics
+    assert tf_core_random_tensors.data.metrics['b'] == 4.0
+    all_a = client.get_metric_history(tf_core_random_tensors.info.run_id, 'a')
+    assert all((x.step - 1) % 4 == 0 for x in all_a)
+
+
+@pytest.fixture
+def tf_estimator_random_data_run():
+    mlflow.tensorflow.autolog()
+    with mlflow.start_run() as run:
+        dir = tempfile.mkdtemp()
+        CSV_COLUMN_NAMES = ['SepalLength', 'SepalWidth', 'PetalLength', 'PetalWidth', 'Species']
+        SPECIES = ['Setosa', 'Versicolor', 'Virginica']
+
+        train_path = tf.keras.utils.get_file(
+            "iris_training.csv", "https://storage.googleapis.com/download"
+                                 ".tensorflow.org/data/iris_training.csv")
+        test_path = tf.keras.utils.get_file(
+            "iris_test.csv", "https://storage.googleapis.com/download"
+                             ".tensorflow.org/data/iris_test.csv")
+
+        train = pd.read_csv(train_path, names=CSV_COLUMN_NAMES, header=0)
+        test = pd.read_csv(test_path, names=CSV_COLUMN_NAMES, header=0)
+
+        train_y = train.pop('Species')
+        test_y = test.pop('Species')
+
+        def input_fn(features, labels, training=True, batch_size=256):
+            """An input function for training or evaluating"""
+            # Convert the inputs to a Dataset.
+            dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))
+
+            # Shuffle and repeat if you are in training mode.
+            if training:
+                dataset = dataset.shuffle(1000).repeat()
+
+            return dataset.batch(batch_size)
+
+        my_feature_columns = []
+        for key in train.keys():
+            my_feature_columns.append(tf.feature_column.numeric_column(key=key))
+
+        feature_spec = {}
+        for feature in CSV_COLUMN_NAMES:
+            feature_spec[feature] = tf.placeholder(dtype="float", name=feature, shape=[150])
+
+        receiver_fn = tf.estimator.export.build_raw_serving_input_receiver_fn(feature_spec)
+
+        classifier = tf.estimator.DNNClassifier(
+            feature_columns=my_feature_columns,
+            # Two hidden layers of 10 nodes each.
+            hidden_units=[30, 10],
+            # The model must choose between 3 classes.
+            n_classes=3,
+            model_dir=dir)
+
+        classifier.train(
+            input_fn=lambda: input_fn(train, train_y, training=True),
+            steps=5000)
```

Comment:
Could we run this for fewer steps & make this a small test? Seems like it'd work to just do like 300 steps